### ISSUE TITLE
[macros] evaluate support for `avoid_unused_constructor_parameters`

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4934]

**Comment by pq** [comment]
The above is now supported w/ https://github.com/dart-lang/sdk/commit/4e3ff1d17d8f36dfde1ff1fbe52e3d2cd21449f7.

The following case is interesting though (thanks @bwilkerson!)

```dart
  test_augmentedConstructor_base() async {
    newFile('$testPackageLibPath/a.dart', r'''
augment library 'test.dart';

augment class A {
  final int y;
  augment A(int x) : y = x;
}
''');

    await assertNoDiagnostics(r'''
import augment 'a.dart';

class A {
  A(int x);
}
''');
  }
```

The rub is that `A(int x);` likely shouldn't produce a lint because in the augmentation, `x` is referenced in the assignment to `y`. 

To get this test to pass though, we'd need to visit the ASTs of the augmentations in `AvoidUnusedConstructorParameters` to ensure that parameters are unused in all of them and currently we only have _elements_ via the `augmented` property of the constructor declaration.

(Note that we'll need to do similar things when evaluating if a field could be `final` or `late` or is unused or...)

/fyi @scheglov
